control-plane-operator/controllers/hostedcontrolplane/configoperator/reconcile.go

```diff
@@ -151,6 +151,51 @@ func ReconcileRole(role *rbacv1.Role, ownerRef config.OwnerRef) error {
 			},
 		},
 	}
+
+	switch platform {
+	case hyperv1.KubevirtPlatform:
+		// By isolating these rules behind the KubevirtPlatform switch case,
+		// we know we can add/remove from this list in the future without
+		// impacting other platforms.
+		role.Rules = append(role.Rules, []rbacv1.PolicyRule{
```

Comment:
do we need to add this perms to the cpo as well i.e. reconcileControlPlaneOperatorRole to honour [privilege escalation prevention](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping)?

Reply:
the cpo role already has these route and service related permissions [1] [2]. I'm only transferring a subset that we need to hcco. 


1. https://github.com/openshift/hypershift/blob/bdb014332d070085d2ed8319a213c7a4c0a05f1f/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L2274
2. https://github.com/openshift/hypershift/blob/bdb014332d070085d2ed8319a213c7a4c0a05f1f/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L2289